components/Templates/includes/auto-template/Pods_Templates_Auto_Template_Front_End.php

```diff
@@ -231,95 +250,182 @@ public function auto_pods() {
+		if ( in_the_loop() ) {
+			return get_post_type();
+		}
 
-		// see if we are on a post type and if so, set $current_post_type to post type
-		if ( isset( $obj->post_type ) ) {
-			$current_post_type = $obj->post_type;
+		// Start by getting current post or stdClass object.
+		$obj = get_queried_object();
 
+		// See if we are on a post type and if so, set $current_post_type to post type.
+		if ( isset( $obj->post_type ) ) {
+			$pod_name = $obj->post_type;
 		} elseif ( isset( $obj->taxonomy ) ) {
-			$current_post_type = $obj->taxonomy;
+			$pod_name = $obj->taxonomy;
 		} elseif ( isset( $obj->name ) ) {
-			$current_post_type = $obj->name;
+			$pod_name = $obj->name;
 		} elseif ( is_home() ) {
-			$current_post_type = 'post';
+			$pod_name = 'post';
 		} else {
-			$current_post_type = false;
+			$pod_name = false;
 		}
 
-		return $current_post_type;
+		return $pod_name;
 	}
 
 	/**
 	 * Outputs templates after the content as needed.
 	 *
 	 * @param string $content Post content
+	 * @param mixed  $obj     Object context
 	 *
 	 * @uses  'the_content' filter
 	 *
 	 * @return string Post content with the template appended if appropriate.
 	 *
 	 * @since 2.4.5
 	 */
-	public function front( $content ) {
+	public function front( $content, $obj = null ) {
+
+		// The current pod type.
+		$pod_type = '';
 
-		// get the current post type
-		$current_post_type = $this->current_post_type();
+		// Get the current pod name.
+		$pod_name = $this->get_pod_name();
 
-		// now use other methods in class to build array to search in/ use
+		// Now use other methods in class to build array to search in/ use.
 		$possible_pods = $this->auto_pods();
 
-		// check if $current_post_type is the key of the array of possible pods
-		if ( isset( $possible_pods[ $current_post_type ] ) ) {
-			// get array for the current post type
-			$this_pod = $possible_pods[ $current_post_type ];
+		if ( $obj ) {
+			if ( $obj instanceof WP_Post ) {
+				$pod_type = 'post';
+				$pod_name = $obj->post_type;
+				$pod_id   = $obj->ID;
+			} elseif ( $obj instanceof WP_Term ) {
+				$pod_type = 'taxonomy';
+				$pod_name = $obj->taxonomy;
+				$pod_id   = $obj->term_id;
+			} elseif ( $obj instanceof WP_Comment ) {
+				$pod_type = 'comment';
+				$pod_name = 'comment';
+				$pod_id   = $obj->comment_ID;
+			} else {
+				$obj = null;
+			}
+		}
 
-			$filter = $this->get_pod_filter( $current_post_type, $possible_pods );
+		if ( ! $pod_type ) {
+			// Build Pods object for current item.
+			$pod_id = get_queried_object_id();
+			if ( is_singular() || in_the_loop() ) {
+				$pod_type = 'post';
+			} elseif ( is_tax() ) {
+				// Outside the loop in a taxonomy, we want the term.
+				$obj      = get_queried_object();
+				$pod_type = 'taxonomy';
+				$pod_name = $obj->taxonomy;
+			} else {
+				// Backwards compatibility.
+				global $post;
+				if ( $post ) {
+					$pod_id   = $post->ID;
+					$pod_type = 'post';
+				}
+			}
+		}
+		// @todo Users?
```

Comment:
May also be a good idea to do users too

Reply:
We might be able to hook somewhere into the author archive and output before or after the loop there.